### PR TITLE
Faster merge

### PR DESCRIPTION
**Pull request by mariane-bekker**

### CODE REVIEW

**Comment by mariane-bekker** [comment]
Did some time comparison... this merge saves a millisecond or more for certain requests... it might make a difference especially for millions of requests

File: test/lib/util/merge_test.js [hunk 1]
```diff
@@ -211,7 +256,7 @@ describe('util: merge', function() {
             [ { foo: 'foo' }, null ],
             [ 1, NaN, 2, undefined, 0.4, true, 'foo', Infinity ]
         ].forEach(function(args) {
-            var expectedResult = args[args.length - 1];
+            var expectedResult = args[0];
```
Comment (mariane-bekker) [inline test/lib/util/merge_test.js]: @djvirgen is it okay if now it defaults to the target value??

Comment (djvirgen) [inline test/lib/util/merge_test.js]: No, the merge should replace values from right to left.

```
merge('a', 'b', 'c'); // should return 'c'
```

File: lib/util/merge.js [hunk 2]
```diff
@@ -10,6 +10,14 @@ var debug = require('debug')('monocle-api:util:merge');
  * @return merged object/array
  */
 var merge = module.exports = function(/* obj1, obj2, obj3 */) {
+    function isObject(element) {
```
Comment (djvirgen) [inline lib/util/merge.js]: This should be moved outside of the `merge` function, because it's re-declared for every merge.

**Comment by djvirgen** [comment]
🍒

**Pull request merged**